deploy_pulp3/pulp3_db/tasks/main.yml

```diff
@@ -0,0 +1,34 @@
+- name: Install postgres packages
+  package: name={{ item }} state=present
+  with_items:
+      - postgresql
+      - postgresql-server
+      # needed by postgres-related tasks and other things
+      # using python outside of a virtualenv
+      - python3-psycopg2
+
+- name: Set up postgres data dir
+  command: postgresql-setup --initdb
+  args:
+      creates: /var/lib/pgsql/data/base
+
+- name: Set up postgress access rules
+  copy:
+      src: pg_hba.conf
+      dest: /var/lib/pgsql/data/pg_hba.conf
+      owner: postgres
+      group: postgres
+      mode: 0600
+
+- name: Start and enable postgresql
+  service: name=postgresql state=restarted enabled=yes
```

Comment:
Same question about idempotency. Could this be wrapped in a conditional?

Reply:
honestly saying i do not see much or harm in doing the restart here, i do not know but i guess if there is a change in the pg_hba.conf then we need to restart the service, that means it  forces me to check not just the first task but also the second

Reply:
Is this restart harmful? No. But again, idempotency is a desirable property of Ansible playbooks.

Handlers might be an appropriate solution to the problem of needing to check two tasks:

> These ‘notify’ actions are triggered at the end of each block of tasks in a play, and will only be triggered once even if notified by multiple different tasks.

Reply:
A handler would not be appropriate here, since the postgresql has to be running before database and user can use it. I would suggest `state: started`

Reply:
Of course. Handlers *alone* wouldn't be appropriate. But triggering a restart of this service with a handler may be appropriate.